jdaviz/core/helpers.py

```diff
@@ -274,96 +275,88 @@ def get_model_parameters(self, models=None, model_label=None, x=None, y=None):
-        appear inline, as only lab has a mechanism to have multiple tabs.
-        See Also
-        --------
-        show_in_new_tab
-        show_inline
+        If "sidecar" or "new jupyter tab" modes are called in the "classic"
+        Jupyter notebook, the app will appear inline, as only lab has a mechanism
+        to have multiple tabs.
         """
-        if 'title' not in kwargs:
-            kwargs['title'] = self.app.config
+        try:
+            if mode == "sidecar":
+                if 'title' not in kwargs:
+                    kwargs['title'] = self.app.config
 
-        scar = Sidecar(**kwargs)
-        with scar:
-            display(self.app)
+                scar = Sidecar(**kwargs)
+                with scar:
+                    display(self.app)
 
-        return scar
-    show_in_sidecar.__doc__ = show_in_sidecar.__doc__.format(
-        sidecar_anchor_values=repr(Sidecar.anchor.values)[1:-1])
+                return scar
 
-    def show_in_new_tab(self, **kwargs):
-        """
-        Display the Jdaviz application in a new tab in JupyterLab.
+            elif mode == "new jupyter tab":
+                if 'anchor' in kwargs:
+                    if 'tab' not in kwargs['anchor']:
+                        raise ValueError('new jupyter lab cannot have a non-tab anchor')
+                else:
+                    kwargs['anchor'] = 'tab-after'
+                return self.show(mode="sidecar", **kwargs)
 
-        Additional keywords not listed here are passed into the
-        ``sidecar.Sidecar`` constructor. See
-        `jupyterlab-sidecar <https://github.com/jupyter-widgets/jupyterlab-sidecar>`_
-        for the most up-to-date options.
+            elif mode == "new browser tab":
+                raise NotImplementedError
 
-        Parameters
-        ----------
-        title : str, optional
-            The title of the sidecar tab.  Defaults to the name of the
-            application; e.g., "specviz".
+            elif mode == "popout":
+                raise NotImplementedError
 
-        Returns
-        -------
-        sidecar
-            The ``sidecar.Sidecar`` object used to create the tab.
+            elif mode == "inline":
+                display(self.app)
 
-        Notes
-        -----
-        If this method is called in the "classic" Jupyter notebook, the app will
-        appear inline, as only lab has a mechanism to have multiple tabs.
+            else:
+                raise ValueError("Unrecognized display mode: " + str(mode))
 
-        See Also
-        --------
-        show_in_sidecar
-        show_inline
-        """
-        if 'anchor' in kwargs:
-            if 'tab' not in kwargs['anchor']:
-                raise ValueError('show_in_new_tab cannot have a non-tab anchor')
-        else:
-            kwargs['anchor'] = 'tab-after'
-        return self.show_in_sidecar(**kwargs)
+        except Exception as e:
```

Comment:
might be nice to catch particular exceptions here (since the try/except wraps a lot more code than just the lines that do the displaying).

Reply:
So I left it generic because I effectively wanted any errors to be thrown back to the user, but to always fallback to the IPython display. I figured if anything errors out, the user can be responsible for trying to fix it. Unless there are specific cases you're thinking that should be caught here?

Reply:
I guess right now there isn't much else that could raise an error in that block.  My main concern would be any error introduced by us to the code in the future, or a ctrl+c type exception (although that seems quite unlikely since nothing should ever hang or take significant time).  My second concern is if warnings are hidden/ignored, then it might be difficult for the user to know why it's not behaving as expected - especially in the "unrecognized display mode" case where a simple typo could trigger that.  I'm curious what others think though.

Reply:
Yeah, I suspect I'm in the minority on preferring errors to warnings.  So I won't insist on changes if this was by design.

Reply:
Ironically, I think I'm actually on your side here. I originally caught the error, fell back on the IPython display, THEN reraised the exception to the user to be more explicit. It was just... distasteful in how it appeared. Pey Lian and I had a LONG thread on this last Friday... If you can find a way to better represent it in a way I coudn't, please do!

Reply:
ok, then let's leave it as-is if no one else has objections/suggestions.

Reply:
To be clear, and perhaps I wasn't being clear in my original communication, my original suggestion was to just error out with the exception (i.e., no try-except). However, Duy wanted to fall back to IPython, so in that case I suggested turning exception into warning and then do the fallback. It just does not make sense to me to throw and exception and then fallback to IPython. Perhaps I have missed something here.